### ISSUE TITLE
Should all playbook arrays always be defined as nullable?

### ISSUE THREAD

**Issue by NikolaSivkov** [ansible/schemas#414]
I have the following issue and question:

Given the following playbook
```yaml
---
- name: Fixture
  hosts: localhost
  vars:
  tasks:
```
Currently `ansible-lint` fails with

```
{'name': 'Fixture', 'hosts': 'localhost', 'vars': None, 'tasks': None} is not valid under any of the given schemas
```

However, if we remove `vars` or initialize it as an empty array `vars: []`, the error goes away.
The same however is not true for `tasks` which is inconsistent behavior, given that ansible itself doesn't fail in either case. 

Which brings the question, can we make all playbook arrays nullable?

**Comment by ssbarnea** [comment]
@bcoca @sivel @cidrblock WDYT about this? Should we allow nulls for these?

IMHO, we should not, even if ansible might happen to allow it. Main reason for that is that presence of `null` is almost always a copy/paste or indentation error in YAML.

**Comment by NikolaSivkov** [comment]
>There is no reason the author should create null entries for these. 

I was experimenting with things, and commented  out the items inside the block (and few other things at the same time), but not the block itself. 
Then ansible lint started complaining with a non-sensical error, that I spent a good chunk of time trying to figure out what it meant and what was causing it. And down the rabbit-hole I went.

My opinion is:  
If ansible treats `undefined`, `null` and `[]` the same way for all pre-defined system properties that are arrays, there's no reason the spec should differ from the behavior. 
From a technical point of view, I _know_ there's a difference between the 3 options. And I get why you might not want to treat the 3 options the same way. 
However, speaking as a casual Ansible user, I see no benefit or difference in treating all 3 options differently, because in all 3 cases the result is the same: "nothing to do here".   
Right now, I see 3 choices
1) Leave the spec as it is, inconsistent.
2) Make the spec consistent by defining the arrays as either of 3 states
3) Make the spec consistent by defining the arrays as either `undefined` or an `initialized empty array`

Option 3) will (probably) see at least some complaints and might break things for people.
Option 2) will mirror the actual Ansible behavior and be consistent with that.
